pkg/fqdn/name_manager_test.go

```diff
@@ -147,3 +148,46 @@ func TestNameManagerMultiIPUpdate(t *testing.T) {
 	nameManager.Unlock()
 
 }
+
+func TestActiveConnectionsTTLUpdate(t *testing.T) {
+	now := time.Now()
+	dnsZombies := NewDNSZombieMappings(defaults.ToFQDNsMaxDeferredConnectionDeletes, defaults.ToFQDNsMaxIPsPerHost)
+	sharedIP := netip.MustParseAddr("1.1.1.1")
+	dnsZombies.Upsert(now, sharedIP, "cilium.io")
+
+	var (
+		activeConnectionTTL = int(2 * time.Second)
+		nameManager         = NewNameManager(Config{
+			MinTTL:               int(1 * time.Second),
+			Cache:                NewDNSCache(0),
+			ActiveConnectionsTTL: activeConnectionTTL,
+			IPCache:              testipcache.NewMockIPCache(),
+
+			UpdateSelectors: func(ctx context.Context, selectorIPMapping map[api.FQDNSelector][]netip.Addr, _ uint64) *sync.WaitGroup {
+				return &sync.WaitGroup{}
+			},
+			GetEndpointsDNSInfo: func(_ string) []EndpointDNSInfo {
+				return []EndpointDNSInfo{
+					{
+						ID:         "1",
+						DNSHistory: NewDNSCache(0),
+						DNSZombies: dnsZombies,
+					},
+				}
+			}})
+	)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// GC should return 1 zombie with updated TTL
+	nameManager.GC(ctx)
+	for _, v := range nameManager.cache.forward {
+		for _, vv := range v {
+			expectedHr, expectedMin, expectedSec := time.Now().Add(time.Duration(activeConnectionTTL) * time.Second).Clock()
```

Comment:
There's still something I don't understand about the use of time here. Specifically, `time.Duration(activeConnectionTTL)` is getting multiplied by `time.Second`, but `activeConnectionTTL` is defined as `int(2*time.Second)`. Duration is measured in nanoseconds, so `activeConnectionsTTL` would be 2e9, then the value of the parameter to `Add()` here would be 2e18 :thinking: . How is this then leading to the correct value for the expiration offset?

Reply:
It'd be nice if we can drop the casts from this code to simplify (may depend on changing some units of some fields).

Nit: `time.Now()` here should probably also rely on the same `now` value pulled at the start of the test rather than fetching a fresh "now" value. It's unlikely, but if there is a significant scheduling delay between `now` fetched at the start of the test and used to calculate timers vs. `time.Now()` at this point in the test, then it could cause the test to become unreliable due to reliance on wall clock + expecting the code between to take zero time.

Reply:
I see. You're right. If I add a debug point after `time.now()`, the test case won't pass because of the delay.
Let me use the same now().

Also, let me add a brief description here about the test case I am trying to achieve and what it is doing. Will update it this week.